it/controllers/pensions/annualAllowance/PensionProviderTaxPaidAnnualAllowanceControllerISpec.scala

```diff
@@ -0,0 +1,488 @@
+            val pensionUserData = pensionsUserDataWithAnnualAllowances(pensionAnnualAllowanceViewModel, isPriorSubmission = false)
+
+            lazy val result: WSResponse = submitPage(user, pensionUserData, form)
+
+            "has the correct status" in {
+              result.status shouldBe BAD_REQUEST
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+            h1Check(user.specificExpectedResults.get.expectedHeading)
+            captionCheck(expectedCaption(taxYearEOY), captionSelector)
+            radioButtonCheck(yesText, 1, checked = Some(true))
+            radioButtonCheck(noText, 2, checked = Some(false))
+            buttonCheck(buttonText, continueButtonSelector)
+            textOnPageCheck(poundPrefixText, poundPrefixSelector)
+            inputFieldValueCheck(amountInputName, amountInputSelector, "")
+            textOnPageCheck(totalNonUkTax, amountText)
+            errorAboveElementCheck(user.specificExpectedResults.get.noEntryAmountErrorMessage)
+            formPostLinkCheck(pensionProviderTaxPaidAnnualAllowanceUrl(taxYearEOY), formSelector)
+          }
+
+          "returns error message when user enters yes and enter 0 for amount and click continue button" which {
+
+            lazy val form: Map[String, String] = Map(
+              RadioButtonAmountForm.yesNo -> RadioButtonAmountForm.yes, RadioButtonAmountForm.amount2 -> "0")
+            val pensionAnnualAllowanceViewModel = aPensionAnnualAllowanceViewModel.copy(
+              pensionProvidePaidAnnualAllowanceQuestion = Some(true),
+              taxPaidByPensionProvider = Some(BigDecimal(existingAmount)))
+            val pensionUserData = pensionsUserDataWithAnnualAllowances(pensionAnnualAllowanceViewModel, isPriorSubmission = false)
+
+            lazy val result: WSResponse = submitPage(user, pensionUserData, form)
+
+            "has the correct status" in {
+              result.status shouldBe BAD_REQUEST
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            h1Check(user.specificExpectedResults.get.expectedHeading)
+            captionCheck(expectedCaption(taxYearEOY), captionSelector)
+            radioButtonCheck(yesText, 1, checked = Some(true))
+            radioButtonCheck(noText, 2, checked = Some(false))
+            buttonCheck(buttonText, continueButtonSelector)
+            textOnPageCheck(poundPrefixText, poundPrefixSelector)
+            inputFieldValueCheck(amountInputName, amountInputSelector, zeroAmount)
+            textOnPageCheck(totalNonUkTax, amountText)
+            errorAboveElementCheck(amountZeroErrorMessage)
+            formPostLinkCheck(pensionProviderTaxPaidAnnualAllowanceUrl(taxYearEOY), formSelector)
+          }
+
+          "returns error message when user enters above 100,000,000,000 for amount and click continue button" which {
```

Comment:
Suggest rename to "returns error message when user enters above the maximum amount and clicks continue button"

Reply:
test case name updated as mentioned in the comment

Reply:
Changed.